source/_docs/guides/jenkins.md

```diff
@@ -0,0 +1,310 @@
+
+    ```nohighlight
+    #!/bin/bash
+    echo "Logging into Terminus"
+    terminus auth:login --machine-token=${TERMINUS_TOKEN}
+    ```
+
+3. Verifies the dev site is awake and in git mode. Note that these are separate build steps:
+
+    ```nohighlight
+    echo "Waking Dev environment."
+    terminus env:wake -n ${SITE_ID}.dev
+    ```
+
+    ```nohighlight
+    echo "Setting site to git mode."
+    terminus connection:set ${SITE_ID}.dev git
+    ```
+
+4. Jenkins creates a Multidev and pushes the new code to this environment
+
+    ```nohighlight
+    echo "Creating multidev"
+    cd ${WORKSPACE}
+    terminus build:env:create ${SITE_ID}.dev ci-${BUILD_ID} --yes
+    ```
+
+    ```nohighlight
+    echo "Run database updates and clear cache"
+    terminus drush -n ${SITE_ID}.ci-${BUILD_ID} -- updatedb -y
+    terminus drush ${SITE_ID}.ci-${BUILD_ID} cr
+    ```
+
+5. Then the test suite we include with the example is run.
+
+    ```nohighlight
+    echo "Running behat"
+    TERMINUS_ENV=ci-$BUILD_ID TERMINUS_SITE=$SITE_ID $WORKSPACE/tests/scripts/run-behat
+    ```
+
+6. Add one **Conditional step (single)** build task, to merge the code from the Pantheon Multidev to the pantheon/master, i.e. your dev site on Pantheon. This will only happen when changing the master branch.
+
+    - Under **Run?** select **Regular expression match**.
+
+        - **Expression**: `(?i).*origin/master.*`
+
+        - **Label**: `${ENV,var="GIT_BRANCH"}`
+
+    - Under **Builder** select **Execute Shell**
+
+        - Command:
+
+                echo "Merging multi-dev changes to master"
+                terminus build:env:merge -n ${SITE_ID}.ci-${BUILD_ID} --yes
+
+    Your conditional step should look like this:
+    ![Conditional Step](/source/docs/assets/images/integrations/jenkins-conditional.png)
+
+7. Finally, a cleanup task:
+
+    ```nohighlight
+    echo "Cleaning up multidev & branches"
+    git -C ${WORKSPACE} remote remove pantheon
+    git -C ${WORKSPACE} remote prune origin
+    terminus build:env:delete:ci ${SITE_ID} --keep=2  --yes
+    ```
+
+### Add Post-build Actions
+
+Under **Post-build Actions** is another button labelled **Add post-build action**. Click on it and select the option "Set GitHub commit status (universal)". In the fields that appear, choose the following options:
+
+ - **Commit SHA**: "Latest build revision".
+
+ - **Repositories**: "Any defined in job repository".
+
+ - **Commit context**: "From GitHub property with fallback to job name".
+
+ - **Status Result**: "One of default messages and statuses".
+
+ - **Status backref**: "Backref to the build"
+
+Finally, hit **Save** to complete the configuration of your Jenkins build process.
+
+Now on your local computer, create a new branch and makes a change to it. When you commit and push to GitHub, a build should initiate. You should see the results of the test, which link to the Jenkins job:
+
+![Passing Github test](/source/docs/assets/images/integrations/test_pass.png)
```

Comment:
The image here should expand the "show all checks" link to show jenkins test and to include some markup to point out what element on the page we're referring to.